packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -142,109 +94,493 @@ describe("Decentraland", () => {
+            expiresAt,
+          } = landForSale;
+
+          checkAsset(land, mana, landForSale, ephemeralAddress);
+
+          await confirmBalances(land, [ephemeralAddress], [0]);
+
+          // LANDRegistry contract doesn't implement getFingerprint function
+          const fingerprint = "0x";
+          marketplace.setWallet(ephemeralWallet);
+          const executeOrderAction = marketplace.safeExecuteOrder(
+            nftAddress,
+            `${assetId}`,
+            `${priceInWei}`,
+            `${fingerprint}`,
+            gasParams
+          );
+
+          await executeOrderAction.waitForNonceToUpdate();
+
+          await confirmBalances(land, [ephemeralAddress], [1]);
+        });
+      });
 
-        await executeOrderAction.waitForNonceToUpdate();
+      // WIP: Not working because of gas issue on Marketplace.safeExecuteOrder() call
+      describe.skip("Using a Gnosis Safe wallet", () => {
+        beforeEach("onboarding", async () => {
+          // Funding ephemeral account with some ethers to pay for gas
+          // TODO: ephemeralWallet should broadcast this action
+          const toAddress = ephemeralAddress;
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
+          const transferEthersAction = gnosisSafe.transferEther(
+            toAddress,
+            SMALL_AMOUNT
+          );
+          await transferEthersAction.waitForNonceToUpdate();
+          await confirmEtherBalances(provider, [toAddress], [SMALL_AMOUNT]);
+
+          // Gnosis Safe should approve Marketplace to spend its MANA Tokens
+          // TODO: ephemeralWallet should broadcast this action
+          const contractAddress = mana.getAddress();
+          const contractABI = mana.getABI();
+          const functionName = "approve";
+          const argsArray = [MARKETPLACE_ADDRESS, manaAmountForShopping];
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
+          const approvalAction = gnosisSafe.customContractAction(
+            contractAddress,
+            contractABI,
+            functionName,
+            argsArray
+          );
+          await approvalAction.waitForNonceToUpdate();
+
+          const owner = GNOSIS_SAFE_ADDRESS;
+          const spender = MARKETPLACE_ADDRESS;
+          const allowance = await mana.allowance(owner, spender);
+          expect(`${allowance}`).to.equal(`${manaAmountForShopping}`);
+        });
+
+        it("should buy an estate", async () => {
+          const {
+            assetId,
+            nftAddress,
+            seller,
+            priceInWei,
+            expiresAt,
+          } = estateForSale;
+
+          const buyerAddress = GNOSIS_SAFE_ADDRESS;
+          await checkAsset(estate, mana, estateForSale, buyerAddress);
+
+          await confirmBalances(estate, [GNOSIS_SAFE_ADDRESS], [0]);
+
+          // Gnosis Safe should approve Marketplace to spend its MANA Tokens
```

Comment:
Did you mean to have this comment line here? I think you just meant to copy the line below it. Because safeExecuteOrder is what happens below, not the approval.

Reply:
```suggestion
```

Reply:
@marcelomorgado Maybe accept this one too before we merge

Reply:
I've changed the comment, I'll commit it now.

Reply:
Done.

Reply:
Thanks! Resolving